server/aesdsocket.c

```diff
@@ -0,0 +1,254 @@
+                if(accept_rc==-1)
+                {
+                        syslog(LOG_ERR, "fail to accept");
+                        perror("accept");
+                        exit(6);
+                }
+                else
+                {
+                        syslog(LOG_INFO,"Accepts connection from %s",inet_ntoa(conn_Addr.sin_addr));
+                }
+
+
+                fd = open("/var/tmp/aesdsocketdata", O_APPEND | O_WRONLY); 
+                rec_fd=recv(accept_rc,buffer,1024,0);
+                if(rec_fd==-1)
+                {
+                        syslog(LOG_ERR, "Error in receiving of data packets from client");
+                        perror("recv");
+                        exit(6);
+                }
+
+
+                ptr = (char*)malloc(1);
+                if(ptr==NULL)
+                {
+                        perror("malloc");
+                        exit(14);
+                }
+
+                while(rec_fd!=0) // receiving packets
+                {
+
+                        for(i=0;i<1024;i++)
+                        {
+                                if(buffer[i]=='\n')//checking for newline
+                                {
+                                        i++;
+                                        write_flag=true;
+                                        break;
+                                }
+                        }
+                        bytes_to_write+=i;
+                        ptr=(char *)realloc(ptr,bytes_to_write+1);//reallocating memory
+                        if(ptr==NULL)
+                        {
+                                perror("realloc");
+                                exit(8);
+                        }
+                        memcpy(ptr+bytes_to_write-i,buffer,i);// copying the buffer to pointer
+                        memset(buffer,0,1024);// clearing buffer
+
+
+                        if(write_flag==true)
+                        {
+
+                                wr_bytes= write(fd,ptr,bytes_to_write); //write packets contents to file
+                                if(wr_bytes!=bytes_to_write)
+                                {
+                                perror("write");
+                                exit(8);
+                                }
+
+                                close(fd);
+                                write_flag=false;
+                                total_packet_size += bytes_to_write;
+                                int op_fd=open("/var/tmp/aesdsocketdata",O_RDONLY);//opening file
+                                if(op_fd==-1)
+                                {
+                                        syslog(LOG_ERR, "fail to open");
+                                        perror("open");
+                                        exit(10);
+                                }
+                                char read_arr[total_packet_size];
```

Comment:
@malolasimman what assumption does this code make about the size of the process stack vs the total size of all packets sent?